libr/anal/esil.c

```diff
@@ -3132,6 +3220,7 @@ static void r_anal_esil_setup_ops(RAnalEsil *esil) {
 	OP ("$js", esil_js, 1, 0, OT_UNK);
 	OP ("$r", esil_rs, 1, 0, OT_UNK);
 	OP ("$$", esil_address, 1, 0, OT_UNK);
+	OP ("~", esil_signext, 1, 2, OT_MATH);
```

Comment:
not sure if ~ is the best char for this because it's technically used for NOT in other contexts. but i cant think of a better char. comments?

Reply:
Yea I had this internal debate for a long time and thought about some other possible characters but i think ~ is still the best. Its twos-complement in many languages so I think it makes sense to associate with sign extension. I also don't like that its grep in r2 but a bunch of the other operators are also special in r2 so that doesn't change much. I think the other option is just to have it be "SIGN", which I would be fine with but it would make expressions bulkier.